src/main.js

```diff
@@ -10,20 +10,99 @@ Config.set({
         timeout: parseInt(3000), // 90000 or less
     },
     realmPath: 'alpha', // e.g. 'alpha' or 'root'
-    // tree: process.env.TREE, // e.g. 'sdkAuthenticationTree' or 'Login'
+    tree: 'AJ Choice Login', // e.g. 'sdkAuthenticationTree' or 'Login'
 });
 
+const loginElem = document.getElementById('login-form');
+const continueElem = document.getElementById('continue-form');
+const successElem = document.getElementById('success');
+const errorElem = document.getElementById('error');
+const userPanel = document.getElementById('user');
+const userInfoElem = document.querySelector('#user pre');
+
+async function nextSteps(step) {
+    do {
+        const nextStep = await FRAuth.next(step);
+        console.log('nextStep', nextStep);
+        await handleNextStep(nextStep);
+        step = nextStep;
+    } while (step);
+    return;
+}
+
+function getStage(step) {
+    let stage;
+    const nameCallbacks = step.getCallbacksOfType('NameCallback');
+    const passwordCallbacks = step.getCallbacksOfType('PasswordCallback');
+    const choiceCallbacks = step.getCallbacksOfType('ChoiceCallback');
+    
+    if (nameCallbacks.length && passwordCallbacks.length) {
+        stage = 'UsernamePassword';
+    } else if (choiceCallbacks.length) {
+        stage = 'Choice';
+    } else {
+        stage = undefined;
+    }
```

Comment:
We discussed this on Slack, but just to reiterate, the idea of dedicated stages is falling a bit out of favor to just having a generic callback to component mapping. This makes the source code less fragile to updates to the Journey.